src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -1064,6 +1064,206 @@ def get_bumps(self) -> Optional[BumpsDefinition]:
             pitch=Decimal(str(self.get_setting("vlsi.inputs.bumps.pitch"))),
             cell=self.get_setting("vlsi.inputs.bumps.cell"), assignments=assignments)
 
+    def generate_visualization(self) -> None:
+        """
+        Generate an SVG visualizing the chip.
+        Depending on mode, it will display floorplan (placement constraints for the current hierarchy)
+        and/or bumps (overlaid on top of floorplan) and pad designators (if this tool is a PCBDeliverableTool).
+        Call this from any custom hook (not used by any default flow).
```

Comment:
Do we potentially want an action, that makes it easy to just do this visualization?

Reply:
I think this was a philosophical question that we addressed earlier in this PR; "1) mixed into custom flow as a step" was what was decided to do for now.

We can address whether we want a standalone action and/or add it as default steps in par/pcb actions later.

Reply:
Ok we'll discuss offline.